api/hack/ci/ci-github-release.sh

```diff
@@ -22,119 +22,190 @@ set -euo pipefail
 
-# err can be used to print logs to stderr
-err(){
-  echo "E: $*" >>/dev/stderr
-}
+GITHUB_TOKEN="${GITHUB_TOKEN:-$(cat /etc/github/oauth | tr --delete '\n')}"
+export GITHUB_AUTH="Authorization: token $GITHUB_TOKEN"
+
+# this stops execution when GIT_TAG is not overriden and
+# we are not on a tagged revision
+export GIT_TAG="${GIT_TAG:-$(git describe --tags --exact-match)}"
+export GIT_BRANCH="${GIT_BRANCH:-$(git rev-parse --abbrev-ref HEAD)}"
+export GIT_HEAD="${GIT_HEAD:-$(git rev-parse HEAD)}"
+export GIT_REPO="${GIT_REPO:-kubermatic/kubermatic}"
+export RELEASE_PLATFORMS="${RELEASE_PLATFORMS:-linux-amd64 darwin-amd64 windows-amd64}"
+
+# By default, this script is used to released tagged revisions,
+# for which a matching tag must exist in the dashboard repository.
+export DASHBOARD_GIT_TAG="${DASHBOARD_GIT_TAG:-$GIT_TAG}"
+
+# RELEASE_NAME allows to customize the tag that is used to create the
+# Github release for, while the Helm charts and things will still
+# point to GIT_TAG
+export RELEASE_NAME="${RELEASE_NAME:-$GIT_TAG}"
 
 # utility function setting some curl default values for calling the github API
 # first argument is the URL, the rest of the arguments is used as curl
 # arguments.
 function github_cli {
-  local url=${1}
+  local url="$1"
   curl \
     --retry 5 \
     --connect-timeout 10 \
-    -H "Authorization: token ${GITHUB_TOKEN}" \
-    "${@:2}" "${url}"
+    --header "$GITHUB_AUTH" \
+    "${@:2}" "$url"
 }
 
 # creates a new github release
 function create_release {
-  local tag="${1}"
-  local name="${2}"
-  local prerelease="${3}"
-  data=$(cat << EOF
-{
-  "tag_name": "$tag",
-  "name": "$name",
-  "prerelease": $prerelease
-}
-EOF
-)
+  local tag="$1"
+  local name="$2"
+  local prerelease="$3"
+  local body="${4:-}"
+  local data
+
+  # using named arguments is a nice way to ensure special
+  # characters in the body survive the JSON encoding
+  data="$(
+    jq --null-input \
+       --arg tag_name "$tag" \
+       --arg name "$name" \
+       --argjson prerelease "$prerelease" \
+       --arg body "$body" \
+       '{"tag_name":$tag_name,"name":$name,"prerelease":$prerelease,"body":$body}'
+  )"
+
   github_cli \
-    "https://api.github.com/repos/${repo}/releases" \
-    -f --data "${data}"
+    "https://api.github.com/repos/$GIT_REPO/releases" \
+    --fail --data "$data"
 }
 
 # upload an archive from a file
 function upload_archive {
-  local file="${1}"
+  local file="$1"
   res=$(github_cli \
-    "https://uploads.github.com/repos/$repo/releases/$releaseID/assets?name=${file}" \
-    -H "Accept: application/json" \
-    -H 'Content-Type: application/gzip' \
-    -s --data-binary "@${file}")
-  if echo "${res}" | jq -e '.'; then
-    # it the response contain errors
-    if echo "${res}" | jq -e '.errors[0]'; then
-      for err in $(echo "${res}" | jq -r '.errors[0].code'); do
+    "https://uploads.github.com/repos/$GIT_REPO/releases/$releaseID/assets?name=$(basename "$file")" \
```

Comment:
FYI:

```
$ gh release upload -h
Upload asset files to a GitHub Release.

To define a display label for an asset, append text starting with '#' after the
file name.


USAGE
  gh release upload <tag> <files>... [flags]

FLAGS
  --clobber   Overwrite existing assets of the same name

INHERITED FLAGS
      --help              Show help for command
  -R, --repo OWNER/REPO   Select another repository using the OWNER/REPO format

LEARN MORE
  Use 'gh <command> <subcommand> --help' for more information about a command.
  Read the manual at https://cli.github.com/manual

```

Reply:
Yeah but we don't have it installed in our Docker images. And just for wrapping a single curl call, I don't see much value in adding the gh binary to our Docker images. ^^